src/shared/components/lollipopMutationPlot/LollipopMutationPlot.tsx

```diff
@@ -208,26 +208,45 @@ export default class LollipopMutationPlot extends React.Component<ILollipopMutat
     @computed private get domains():DomainSpec[] {
         if (!this.props.store.pfamDomainData.isComplete ||
             !this.props.store.pfamDomainData.result ||
-            this.props.store.pfamDomainData.result.length === 0)
+            this.props.store.pfamDomainData.result.length === 0 ||
+            !this.props.store.canonicalTranscript.isComplete ||
+            !this.props.store.canonicalTranscript.result ||
+            this.props.store.canonicalTranscript.result.pfamDomains.length === 0)
         {
             return [];
         } else {
-            return this.props.store.pfamDomainData.result.map((domain:PfamDomain)=>{
+            return this.props.store.canonicalTranscript.result.pfamDomains.map((range:PfamDomainRange)=>{
+                const domain = this.domainMap[range.pfamDomainId];
                 return {
-                    startCodon: domain.pfamDomainStart,
-                    endCodon: domain.pfamDomainEnd,
-                    label: domain.pfamDomainName,
-                    color: this.domainColorMap[domain.pfamDomainId],
-                    tooltip: this.domainTooltip(domain)
+                    startCodon: range.pfamDomainStart,
+                    endCodon: range.pfamDomainEnd,
+                    label: domain.name,
+                    color: this.domainColorMap[range.pfamDomainId],
+                    tooltip: this.domainTooltip(range, domain)
                 };
             });
         }
     }
 
     @computed private get domainColorMap(): {[pfamAccession:string]: string}
     {
-        if (!this.props.store.pfamDomainData.isPending && this.props.store.pfamDomainData.result) {
-            return generatePfamDomainColorMap(this.props.store.pfamDomainData.result);
+        if (!this.props.store.canonicalTranscript.isPending && 
```

Comment:
now use canonicalTranscript to make the domain color map instead of using the pfam domain data